### PR TITLE
added awesome-scatter-plot with index extry

### PR DESCRIPTION
**Pull request by MaybeRex**
## Added
- awesome-scatter-plot
- index entry
- example

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
     "js-message": "latest",
     "event-pubsub": "latest",
     "browser-error-classes": "latest",
-    "document-register-element": "latest"
+    "document-register-element": "latest",
+    "highcharts": "^4.2.6"
```
Comment (asida18) [inline bower.json:32]: should be `'latest'`

File: components/charts/awesome-scatter-plot.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        component.create=function createAwesomeScatterPlot() {
+            return class AwesomeScatterPlot extends awesome.component.BaseComponent{
+                createdCallback(){
+                    this.mergeDataset(defaults);
+                    super.createdCallback();
+                    this.classList.add(AwesomeScatterPlot.elementTagName);
+
+                    this.innerHTML=`
+                        <section class = 'chartArea'>
+                        </section>
```
Comment (asida18) [inline components/charts/awesome-scatter-plot.js]: unnecessary section tag. just use `this`

File: components/charts/awesome-scatter-plot.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                    `;
+                    this.chartArea = this.querySelector('.chartArea');
```
Comment (asida18) [inline components/charts/awesome-scatter-plot.js]: this is no longer needed.

File: components/charts/awesome-scatter-plot.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                    this.chart = null;
+
+                    this.chartData = {
+                        chart:{
+                            renderTo: this.chartArea,
```
Comment (asida18) [inline components/charts/awesome-scatter-plot.js]: renderTo should just render to `this`

File: components/charts/awesome-scatter-plot.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                            min: this.dataset.yMin,
+                            max: this.dataset.yMax
+                        }
+                    }
+                }
+
+                load(data){
+                    this.chartData.series = [];
+
+                    for(const dataset in data){
```
Comment (asida18) [inline components/charts/awesome-scatter-plot.js]: give `dataset` a better name

File: examples/awesome-scatter-plot.html [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+                            {
+                                x:Math.floor(Math.random() * 200) - 100,
+                                y:Math.floor(Math.random() * 200) - 100
+                            },{
+                                x:Math.floor(Math.random() * 200) - 100,
+                                y:Math.floor(Math.random() * 200) - 100
+                            }
+                        ],
+                        color:'green',
+                        symbol:'circle'
```
Comment (asida18) [inline examples/awesome-scatter-plot.html]: maybe change symbol for others to show different symbol types

File: bower.json (hunk 1 above)
Comment (MaybeRex) [inline bower.json:32]: good comments @asida18

File: bower.json [hunk 7]
```diff
@@ -28,6 +28,7 @@
     "js-message": "latest",
     "event-pubsub": "latest",
     "browser-error-classes": "latest",
-    "document-register-element": "latest"
+    "document-register-element": "latest",
+    "highcharts": "latest"
```
Comment (RIAEvangelist) [inline bower.json]: dont use latest

for highcharts only

File: components/charts/awesome-scatter-plot.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+                            align: 'right',
+                            verticalAlign: 'top',
+                            layout: 'vertical',
+                            title:{
+                                text:this.dataset.legendTitle
+                            }
+                        },
+                        plotOptions:{
+                            allowPointSelect:true
+                        },
```
Comment (RIAEvangelist) [inline components/charts/awesome-scatter-plot.js:46]: no option to change?

File: components/charts/awesome-scatter-plot.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+                        title:{
+                            text:this.dataset.chartTitle,
+                            align:'center'
+                        },
+                        legend:{
+                            align: 'right',
+                            verticalAlign: 'top',
+                            layout: 'vertical',
+                            title:{
```
Comment (RIAEvangelist) [inline components/charts/awesome-scatter-plot.js:40]: not configurable?

File: components/charts/awesome-scatter-plot.js [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+                                    symbol:data[dataName].symbol
+                                }
+                            }
+                        );
+                    }
+
+                    this.chart = new Highcharts.chart(
+                        this.chartData
+                    );
+                }
```
Comment (RIAEvangelist) [inline components/charts/awesome-scatter-plot.js:84]: is there a destructor for Highcharts?

**Pull request merged**